### PR TITLE
Fixes #46 - Uploading ronex path and ids to the parameter server

### LINKED ISSUE #46: Register ronexes in parameter server
The driver should register in the parameter server:
- ronexes:
  - ronex:
    - product_id_alias (see #45)
    - serial
    - alias (if specified)

### PR DESCRIPTION
**Pull request by ugocupcic**
pull request 55 was closed automatically

### CODE REVIEW

File: sr_ronex_drivers/src/sr_board_mk2_gio.cpp [hunk 1]
```diff
@@ -40,6 +40,17 @@
 
 SrBoardMk2GIO::~SrBoardMk2GIO()
 {
+  //remove parameters from server
+  //@todo should this be slightly cleaner? probably not necessary for now
```
Comment (markpitchless) [inline sr_ronex_drivers/src/sr_board_mk2_gio.cpp]: Should just need to delete the top of the tree, ie "/ronex/" << parameter_id_

Seems to work from command line:

mark@horus:~/ShadowRobot/Ronex/groovy$ rosparam set /foo/bar 23
mark@horus:~/ShadowRobot/Ronex/groovy$ rosparam set /foo/baz 42
mark@horus:~/ShadowRobot/Ronex/groovy$ rosparam list
/foo/bar
/foo/baz
/rosdistro
/roslaunch/uris/host_horus__51231
/rosversion
/run_id
mark@horus:~/ShadowRobot/Ronex/groovy$ rosparam delete /foo
mark@horus:~/ShadowRobot/Ronex/groovy$ rosparam list
/rosdistro
/roslaunch/uris/host_horus__51231
/rosversion
/run_id

**Comment by markpitchless** [comment]
Some integration tests should also be added for this. I guess a minimum is that the test ronexs are listed properly.
(Testing plug and unplug would need an arm and hand for the tests to use ;-)

**Comment by ugocupcic** [comment]
I pushed the new delete top level param. Cool stuff.

Regarding integration tests, I already added some notes to the test issue :)

On 17 September 2013 16:48, Mark Pitchless notifications@github.com wrote:

> Some integration tests should also be added for this. I guess a minimum is
> that the test ronexs are listed properly.
> (Testing plug and unplug would need an arm and hand for the tests to use
> ;-)
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/shadow-robot/sr-ronex-dev/pull/59#issuecomment-24593935
> .

## 

Ugo Cupcic _Senior Software Engineer_+44 20 7700 2487ugo@shadowrobot.com
www.shadowrobot.com

## 

---

_Shadow Robot Company Ltd._
251 Liverpool Road, N1 1LX, UK

## Registered Number 3308007 (England & Wales)

File: sr_ronex_drivers/include/sr_ronex_drivers/ronex_utils.hpp [hunk 2]
```diff
@@ -83,13 +84,68 @@
...
+   * @return the index of the ronex in the parameter server. -1 if not found.
+   *         or the next free index if ronex_id == ""
+   */
+  static inline int get_ronex_param_id(std::string ronex_id)
+  {
+    std::string param;
+
+    bool last_ronex = false;
+    int ronex_parameter_id = 0;
+    while( !last_ronex )
```
Comment (toliver) [inline sr_ronex_drivers/include/sr_ronex_drivers/ronex_utils.hpp]: last_ronex is not used to exit the loop in any case (nothing to worry much about)

File: sr_ronex_drivers/include/sr_ronex_drivers/ronex_utils.hpp [hunk 3]
```diff
@@ -83,13 +84,68 @@
...
+      }
+      else
+      {
+        if( ronex_id.compare("") != 0)
+        {
+          //we were looking for a specific ronex and didn't find it -> return -1
+          return -1;
+        }
+
+        return ronex_parameter_id;
```
Comment (ugocupcic) [inline sr_ronex_drivers/include/sr_ronex_drivers/ronex_utils.hpp:142]: No this is the new index that can be used for the ronex (first available
index)

On 17 September 2013 17:04, toliver notifications@github.com wrote:

> In sr_ronex_drivers/include/sr_ronex_drivers/ronex_utils.hpp:
> 
> > -          if( ronex_id.compare(param) == 0)
> > -          {
> > -            return ronex_parameter_id;
> > -          }
> > -        }
> > -        ++ronex_parameter_id;
> > -      }
> > -      else
> > -      {
> > -        if( ronex_id.compare("") != 0)
> > -        {
> > -          //we were looking for a specific ronex and didn't find it -> return -1
> > -          return -1;
> > -        }
> >   +
> > -        return ronex_parameter_id;
> 
> In this case we are returning ronex_parameter_id but no ronex_id was found
> for this ronex_parameter_id.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/shadow-robot/sr-ronex-dev/pull/59/files#r6407244
> .

## 

Ugo Cupcic _Senior Software Engineer_+44 20 7700 2487ugo@shadowrobot.com
www.shadowrobot.com

## 

---

_Shadow Robot Company Ltd._
251 Liverpool Road, N1 1LX, UK

## Registered Number 3308007 (England & Wales)

**Pull request merged**